src/api/services/adminServiceHelpers.ts

```diff
@@ -80,3 +82,47 @@ export const mapClientTypesToAdminEnums = (
     return adminEnum;
   });
 };
+
+export type ApiKeyDTO = {
+  key_id: string;
+  name: string;
+  contact: string;
+  roles: ApiRoleDTO[];
+  created: number;
+  disabled: boolean;
+  service_id: number;
+};
+
+export type ApiKeyAdmin = Omit<ApiKeyDTO, 'roles'> & { roles: string };
+
+export async function mapApiKeyDTO(adminApiKeys: ApiKeyAdmin[]): Promise<ApiKeyDTO[]> {
```

Comment:
my tired brain is finding this method a bit hard to understand. A few asks:
* Method name - are we mapping from ApiKeyDTO or to ApiKeyDTO? Perhaps we could rename this to something like `mapAdminApiKeysToApiKeyDTOs`?
* Could we possibly extract some of the logic in this to a well-named function to more clearly convey the functionality?

Reply:
Have reffactored it a bit, moved the main extra mapping function out and added a few comments.